artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PageTimedWriter.java

```diff
@@ -118,15 +120,18 @@ public void incrementTask() {
       pendingTasksUpdater.incrementAndGet(this);
    }
 
-   public void addTask(OperationContext context,
+   public int addTask(OperationContext context,
                                     PagedMessage message,
                                     Transaction tx,
-                                    RouteContextList listCtx) {
+                                    RouteContextList listCtx, boolean useFlowControl) {
 
       logger.trace("Adding paged message {} to paged writer", message);
 
+      // the module using the page operation should later call flowControl from this class.
+      // the only exception to this would be from tests where we don't really care about flow control on the TimedExecutor
+      // also: the credits is based on the page size, and we use the encodeSize to flow it.
       int credits = Math.min(message.getEncodeSize() + PageReadWriter.SIZE_RECORD, maxCredits);
-      writeCredits.acquireUninterruptibly(credits);
+
```

Comment:
This line removed acquisition of credit, but the lines immediately below still have what looks like a compensating-release of credit if not started. Should that go now?

Reply:
Oh yeah.. good point!

thanks a lot